examples/aggregator/module/rust/src/lib.rs

```diff
@@ -22,76 +22,117 @@
 //! Clients invoke the module by providing a vector of non-negative numbers and get back an
 //! aggregated vector if an Oak Node has collected more samples than the predefined threshold.
 
-mod aggregator;
+mod data;
 mod proto;
 #[cfg(test)]
 mod tests;
 
-use aggregator::{Monoid, ThresholdAggregator};
-use log::info;
+use aggregator_common::ThresholdAggregator;
+use data::SparseVector;
+use log::{debug, error};
 use oak::grpc;
-use proto::aggregator::Vector;
-use proto::aggregator_grpc::{Aggregator, Dispatcher};
+use proto::aggregator::{Sample, SerializedSparseVector};
+use proto::aggregator_grpc::{Aggregator, AggregatorClient, Dispatcher};
 use protobuf::well_known_types::Empty;
+use std::collections::HashMap;
+use std::convert::{From, TryFrom};
 
-const SAMPLE_THRESHOLD: u64 = 3;
+/// Currently threshold value is hardcoded.
+const SAMPLE_THRESHOLD: u64 = 2;
 
-oak::entrypoint!(oak_main => {
-    oak::logger::init_default();
-    let node = AggregatorNode {
-        aggregator: ThresholdAggregator::<Vector>::new(SAMPLE_THRESHOLD),
-    };
-    Dispatcher::new(node)
-});
+type ThresholdAggregatorMap = HashMap<String, Option<ThresholdAggregator<SparseVector>>>;
```

Comment:
This map seems to obfuscate things instead of making them clearer. I would just inline the extended type below. In any case, could you clarify why there is an `Option` on the RHS?